### ISSUE TITLE
Timeout not respected when chaining calls

### ISSUE THREAD

**Issue by jakob-o** [jcabi/jcabi-http#160]
Hi,

it looks like the timeouts are only respected if set directly before `fetch()` otherwise they get lost, e.g.
```
// OK
Response response = new ApacheRequest(path)
    .method("GET")
    .timeout(connect, read)
    .fetch();
// Timeouts not respected
Response response = new ApacheRequest(path)
    .timeout(connect, read)
    .method("GET")
    .fetch();
```

Regards

Jakob

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this issue

**Comment by yegor256** [comment]
@jakob-o this is _absolutely_ not possible. But let me investigate :)

**Comment by yegor256** [comment]
@jakob-o investigated. It's not possible. Can you please provide a unit test to prove that this really happens.

**Comment by jakob-o** [comment]
@yegor256 See PR #165

**Comment by jakob-o** [comment]
It seems, that neither connect nor read are passed to the constructors in a couple of methods, e.g.
```
        return new BaseRequest(
            this.wire,
            this.home,
            headers,
            this.mtd,
            this.content
        );
```
See:
https://github.com/jcabi/jcabi-http/blob/master/src/main/java/com/jcabi/http/request/BaseRequest.java#L200
https://github.com/jcabi/jcabi-http/blob/master/src/main/java/com/jcabi/http/request/BaseRequest.java#L219

**Comment by yegor256** [comment]
@jakob-o looks like you're right! Many thanks for the PR. Please, fix your branch so that it passes static analysis checks and I will merge.

**Comment by ivanov-d** [comment]
@yegor256 , the bug is still reproducible in the latest version, because fix for back() method from https://github.com/jcabi/jcabi-http/pull/149 was not approved. Shall I open another PR since original author gave up on it?

**Comment by yegor256** [comment]
@jakob-o PR #149 was merged

**Comment by yegor256** [comment]
@rultor release, tag is `0.17.3`

**Comment by yegor256** [comment]
@rultor release, tag is `1.17.3`

**Comment by andreoss** [comment]
@jakob-o Fixed in [1.17.3](https://github.com/jcabi/jcabi-http/releases/tag/1.17.3)

**Comment by 0crat** [comment]
Job `gh:jcabi/jcabi-http#160` is not assigned, can't get performer
